test/over_react/component_declaration/flux_component_test.dart

```diff
@@ -157,13 +163,52 @@ void main() {
     });
 
     test('should not redraw after being unmounted', () async {
-      var renderedInstance = render(TestDefault());
-      TestDefaultComponent component = getDartComponent(renderedInstance);
+      var renderedInstance = render(testComponents.basic());
+      dynamic component = getDartComponent(renderedInstance);
       component.componentWillUnmount();
       component.redraw();
       await nextTick();
       expect(component.numberOfRedraws, equals(0));
     });
+
+    group('calls super in the appropriate lifecycle methods', () {
+      test('componentWillMount', () {
+        expect(() {
+          mount(TestPropValidation()());
+        }, throwsPropError_Required('TestPropValidationProps.required'),
+            reason: 'should have called super, triggering prop validation logic');
+      });
+
+      test('componentWillReceiveProps', () {
+        var jacket = mount((testComponents.propValidation()..required = 'foo')());
+        expect(() {
+          jacket.rerender(testComponents.propValidation()());
+        }, throwsA(anyOf(
+          hasToStringValue('V8 Exception'), /* workaround for https://github.com/dart-lang/sdk/issues/26093 */
+          hasToStringValue(contains('RequiredPropError:')),
+        )),
+            reason: 'should have called super, triggering prop validation logic');
+      });
+
+      test('componentWillUnmount', () {
+        var jacket = mount(testComponents.basic()());
+        dynamic component = jacket.getDartInstance();
+        // Bind this since expectAsync doesn't seem to play well when
+        // called from react-dart's Zone
+        component.getManagedDisposer(Zone.current.bindCallback(
+            expectAsync0(() async {}, reason: 'should have called super, triggering disposal logic')
+        ));
+        jacket.unmount();
```

Comment:
This should be unnecessary - as jackets unmount automatically afaik

Reply:
That's true; I added it in here explicitly since this is testing component unmount, and I thought it would make the test clearer.